src/Relations/BelongsToThrough.php

```diff
@@ -8,8 +8,16 @@
 use Illuminate\Database\Eloquent\Relations\Concerns\SupportsDefaultModels;
 use Illuminate\Database\Eloquent\Relations\Relation;
 use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Database\Query\Expression;
 use Illuminate\Support\Str;
 
+/**
+ * @template TRelatedModel of \Illuminate\Database\Eloquent\Model
+ * @template TIntermediateModel of \Illuminate\Database\Eloquent\Model
+ * @template TDeclaringModel of \Illuminate\Database\Eloquent\Model
+ *
+ * @extends \Illuminate\Database\Eloquent\Relations\Relation<TRelatedModel>
```

Comment:
You're missing the two other generics here

Reply:
You would think so, and that's what I did at first, but specifying all three generics results in:

```
  48     Generic type Illuminate\Database\Eloquent\Relations\Relation<Illuminate\Database\Eloquent\Model,   
         $this(Staudenmeir\BelongsToThrough\IdeHelper\BelongsToThroughRelationsHook),                       
         $this(Staudenmeir\BelongsToThrough\IdeHelper\BelongsToThroughRelationsHook)> in PHPDoc tag @param  
         for parameter $relationship specifies 3 template types, but class                                  
         Illuminate\Database\Eloquent\Relations\Relation supports only 1: TRelatedModel                     
         🪪  generics.moreTypes   
```

Reply:
I added generics to the Relation classes in L11.15, but for older versions of Laravel it gets the generics from the Larastan stubs.

I have an open PR to update Larastan (https://github.com/larastan/larastan/pull/1990), but it hasn't been merged yet. However, I have published the fix on my [fork of Larastan](https://github.com/calebdw/larastan)